src/main/kotlin/org/rust/lang/core/resolve/Processors.kt

```diff
@@ -63,25 +63,20 @@ interface RsResolveProcessorBase<in T : ScopeEntry> {
 
 typealias RsResolveProcessor = RsResolveProcessorBase<ScopeEntry>
 
-fun createProcessor(name: String? = null, processor: (ScopeEntry) -> Boolean): RsResolveProcessor =
-    createProcessorGeneric(name, processor)
-
-fun createProcessor(names: Set<String>?, processor: (ScopeEntry) -> Boolean): RsResolveProcessor =
-    createProcessorGeneric(names, processor)
-
-fun <T : ScopeEntry> createProcessorGeneric(
-    name: String? = null,
-    processor: (T) -> Boolean
-): RsResolveProcessorBase<T> = createProcessorGeneric(name?.let { setOf(it) }, processor)
+fun createStoppableProcessor(processor: (ScopeEntry) -> Boolean): RsResolveProcessor {
+    return object : RsResolveProcessorBase<ScopeEntry> {
+        override fun process(entry: ScopeEntry): Boolean = processor(entry)
+        override val names: Set<String>? = null
+    }
+}
 
-fun <T : ScopeEntry> createProcessorGeneric(
-    names: Set<String>? = null,
-    processor: (T) -> Boolean
-): RsResolveProcessorBase<T> {
-    return object : RsResolveProcessorBase<T> {
-        override fun process(entry: T): Boolean = processor(entry)
-        override val names: Set<String>? = names
-        override fun toString(): String = "Processor(names=$names)"
+fun createProcessor(processor: (ScopeEntry) -> Unit): RsResolveProcessor {
+    return object : RsResolveProcessorBase<ScopeEntry> {
+        override fun process(entry: ScopeEntry): Boolean {
+            processor(entry)
+            return false
+        }
+        override val names: Set<String>? = null
```

Comment:
And in `createStoppableProcessor`

```suggestion
        override val names: Set<String>? get() = null
```